### PR TITLE
feat: Oracles page hints

### LINKED ISSUE #185: Oracles page hints
Display hints on oracles page.

### PR DESCRIPTION
**Pull request by janmichek**
## Description
resolves #185

## Demo
The feature is initiated from develop, hence the tooltips are still red

https://github.com/aeternity/aescan/assets/15363559/04730b3d-9022-4f63-990e-fb75a787c23c

## Checklist:
- [x] add oracle page hints

### CODE REVIEW

File: src/utils/hints/oracles.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export const oraclesHints = {
+  oracleId: 'Identifier of Oracle that is similar to the address of the Account that registered the Oracle. The only difference is the prefix that will be ok_ instead of ak_. This means that each account can only host 1 oracle. It\'s not possible to manage multiple oracles using the same account.',
```
Comment (lukeromanowicz) [inline src/utils/hints/oracles.js:2]: I think most of the oracleId description could be simplified to something like:

_Unique oracle identifier inherited from the account address of the creator._

that implies that there can be only one oracle and that the mentioned hashes are related

File: src/utils/hints/oracles.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+  oracleRegistered: 'To register an Oracle you need to provide a queryFormat and a responseFormat to the registerOracle function of the SDK. In addition to the common transaction options you can provide the oracle specific options queryFee and oracleTtlValue',
```
Comment (lukeromanowicz) [inline src/utils/hints/oracles.js:3]: I think this description is unrelated to what we show, which is simply the keyblock and the date when the registration happened

File: src/utils/hints/oracles.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+  oracleExpiration: 'By default the Oracle will exist for the next 500 Keyblocks. If you intend to keep your oracle running longer you should increase the oracleTtlValue and/or set up a service that automatically extends the TTL before it expires.',
+  queryFee: 'Fee for querying Oracle. QueryFee is optional and defaults to 30000',
```
Comment (lukeromanowicz) [inline src/utils/hints/oracles.js:5]: missing unit next to 30000

**Review by lukeromanowicz — CHANGES_REQUESTED** [review]
I still stand by implementing it using one of the i18n libraries due to the long-term benefits, but I will approve the current form too if that's the dominant opinion in the team. At least keeping it in this form will make it easy to refactor.

Regarding the code itself - clean and nice implementation - as always.

I proposed some improvements in the hint messages themselves but I'm far from an expert in this field, so treat it more like a guess/hint than something certain.

File: src/utils/hints/oracles.js (hunk 1 above)
Comment (marc0olo) [reply src/utils/hints/oracles.js:2]: I like the proposal of @lukeromanowicz 👍

File: src/utils/hints/oracles.js (hunk 2 above)
Comment (marc0olo) [reply src/utils/hints/oracles.js:3]: yes, it is just the moment of time where the oracle has been registered via `OracleRegisterTx`

File: src/utils/hints/oracles.js (hunk 1 above)
Comment (thepiwo) [reply src/utils/hints/oracles.js:2]: 👍

File: src/utils/hints/oracles.js (hunk 2 above)
Comment (thepiwo) [reply src/utils/hints/oracles.js:3]: I agree, we should just summarise here what the meaning/implication of registered is

File: src/utils/hints/oracles.js (hunk 3 above)
Comment (thepiwo) [reply src/utils/hints/oracles.js:5]: "querying Oracle" --> "querying an/the Oracle"

**Review by thepiwo — COMMENTED** [review]
in general I'd question the wording of addressing the user directly as "you" or "your oracle" I think we should be generically explaining the technicalities without addressing users

File: src/utils/hints/oracles.js (hunk 1 above)
Comment (janmichek) [reply src/utils/hints/oracles.js:2]: Thanks, accepted

File: src/utils/hints/oracles.js (hunk 2 above)
Comment (janmichek) [reply src/utils/hints/oracles.js:3]: changed to
`Date, time, and block height when the Oracle has been registered via OracleRegisterTx.`

File: src/utils/hints/oracles.js (hunk 3 above)
Comment (janmichek) [reply src/utils/hints/oracles.js:5]: Both fixed

**Comment by janmichek** [comment]
> in general I'd question the wording of addressing the user directly as "you" or "your oracle" I think we should be generically explaining the technicalities without addressing users

Agree, rephrased.

File: src/utils/hints/oracles.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+export const oraclesHints = {
+  oracleId: 'Unique Oracle identifier inherited from the account address of the creator.',
+  oracleRegistered: 'Date, time, and block height when the Oracle has been registered via OracleRegisterTx.',
```
Comment (thepiwo) [inline src/utils/hints/oracles.js:3]: also here, "Estimated date, time and block..."

File: src/utils/hints/oracles.js [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+  oracleExpiration: 'By default the Oracle will exist for the next 500 Keyblocks. To keep an Oracle running longer the oracleTtlValue should be increased and/or set up a service that automatically extends the TTL before it expires.',
```
Comment (thepiwo) [inline src/utils/hints/oracles.js:4]: oracleTtlValue reads weird

File: src/utils/hints/oracles.js (hunk 4 above)
Comment (janmichek) [reply src/utils/hints/oracles.js]: You are right, fixed

File: src/utils/hints/oracles.js (hunk 5 above)
Comment (janmichek) [reply src/utils/hints/oracles.js]: fixed

**Review by thepiwo — APPROVED** [review]

**Review by csiarab — APPROVED** [review]

File: src/utils/hints/oracles.js [hunk 6]
```diff
@@ -0,0 +1,6 @@
+export const oraclesHints = {
+  oracleId: 'Unique Oracle identifier inherited from the account address of the creator.',
+  oracleRegistered: 'Estimated date, time, and block height when the Oracle has been registered via OracleRegisterTx.',
+  oracleExpiration: 'By default the Oracle will exist for the next 500 Keyblocks. To keep an Oracle running longer the oracleTtl value, it should be increased and/or set up a service that automatically extends the TTL before it expires.',
+  queryFee: 'Fee for querying an Oracle. QueryFee is optional and defaults to 30000 aettos',
```
Comment (michele-franchi) [inline src/utils/hints/oracles.js:5]: Missing full stop, could you add it?

**Review by michele-franchi — APPROVED** [review]
Just missing a full stop but looks good 👏

File: src/utils/hints/oracles.js (hunk 6 above)
Comment (janmichek) [reply src/utils/hints/oracles.js]: fixed

**Comment by janmichek** [comment]
@marc0olo All fixed. Could you check againhere, please?
